docs/design/Network Topology Aware Scheduling.md

```diff
@@ -0,0 +1,575 @@
+![network topology usecase3](images/network-topology-aware/network-topology-usecase3.png)
+
+4. The pods belonging to the same group are required to be scheduled to HCCS topology zone.  
+5. Prefer to schedule the pod group to one HCCS topology zone, if not enough nodes, try to schedule to the RoCE topology zone and then the VPC topology zone.  
+6. Gang scheduling is required in this case, to make sure all pods are able to proceed with their work.
+
+# Scope:
+
+In Scope:
+
+* support clos network topology define and management  
+* support network topology aware scheduling for Volcano job  
+* support spine-leaf network
+
+# Function Detail
+
+### network topology management
+
+Option 1: Describe network topology by labels, there's a proposal in upstream  
+Cons: 
+* complicated to construct tree for the topology information  
+
+Option 2: Describe network topology by CRD, NetworkTopology in Volcano → plugin/scheduling
+
+- label \-\> NetworkTopology   
+- Rest API (NV, HW) \-\> NetworkTopology
+
+Pros: 
+  * easier for debugging:  
+  * engineers don’t need to construct the tree of topology info by themselves.   
+  * the scheduler uses the same as engineers can see.
+
+* Components besides scheduler are able to use the CR as well
+
+#### network topology definition
+
+`HyperNode` is a performance domain which consists of a group of nodes or sub-performance domains.The network bandwidth and latency is the same in one HyperNode. This CRD is used to describe the network topology in Kubernetes cluster.
+
+`Tier` is a way to distinguish different performance domains. The bandwidth and latency are the same in one tier. The smaller the value of the tier, the higher the bandwidth. For example, compute-network and storage network can be in different tiers, or in the compute-network, there are several levels of spine, leaf switches, each level can be identified as a tier.
+
+```go
+type HyperNode struct {
+	metav1.TypeMeta `json:”,inline”`
+	metav1.ObjectMeta `json:”metadata, omitempty”`
+
+	Spec HyperNodeSpec `json:”spec”`
+	Status HyperNodeStatus `json:”status”`
+}
+
+type HyperNodeSpec struct {
+	tier string	`json:"tier,omitempty"`
+	members []MemberSpec	`json:"members,omitempty"`
+}
+
+type MemberSpec struct {
+	name string	`json:"name,omitempty"`
+	type string	`json:"type,omitempty"`
+}
+
+type HyperNodeStatus struct {
+	……
+}
+
+```
+
+Here are two examples:
+
+Example 1:  
+Here is a spine-leaf tree, there are three levels of switches, connecting the 8 nodes.
+![network topology example1](images/network-topology-aware/network-topology-example1.png)
+
+```go
+version: topology.volcano.sh/v1alpha1
+kind: HyperNode
+name: s0
+spec:
+   tier: 1
+   members:
+    name: node0
+    name: node1
+
+version: topology.volcano.sh/v1alpha1
+kind: HyperNode
+name: s1
+spec:
+  HyperNode
```

Comment:
There are redundant `HyperNode` here need to be removed

Reply:
done